h2o-data-rust/src/lib.rs

```diff
@@ -0,0 +1,35 @@
+mod generators;
+mod helpers;
+
+use crate::generators::{GroupByGenerator, RowGenerator};
+use crate::helpers::generate_csv;
+use crate::helpers::pretty_sci;
+
+use kdam::{tqdm, BarExt};
+use pyo3::prelude::*;
+
+#[pyfunction]
+fn generate_groupby_csv(n: u64, k: u64, nas: u8, seed: u64) -> () {
+    let output_name = format!("G1_{}_{}_{}_{}.csv", pretty_sci(n), pretty_sci(n), k, nas);
+    let mut pb = tqdm!(total = n as usize, position = 0);
```

Comment:
If it is a standalone function, then we do not need to specify position of the bar manualy

Reply:
how would you recommend I call the generate_csv() rust function with no pb parameter and not change the other calls in main.rs? I thought about passing the pb as an option but that requires me to change the type for the main.rs calls as well. 

Also I'm assuming that checking if pb is some will slow the write loop down. 

For example:

```rust
#[pyfunction]
fn generate_groupby_csv(output_name: String, n: u64, k: u64, nas: u8, seed: u64) -> () {
    let mut pb: Option<Bar> = None;
    generate_csv(
        &mut GroupByGenerator::new(n, k, nas, seed),
        &output_name,
        &mut pb,
        n,
    );
}
```


```rust
pub fn generate_csv(
    generator: &mut dyn RowGenerator,
    file_name: &str,
    pb: &mut Option<Bar>,
    n_rows: u64,
) -> () {
    let _ = fs::write(PathBuf::from(&file_name), generator.get_csv_header());
    let file = fs::OpenOptions::new()
        .append(true)
        .write(true)
        .open(file_name)
        .unwrap();

    let mut writer = BufWriter::new(file);
    for _ in 0..n_rows {
        writer
            .write(generator.get_csv_row().as_bytes())
            .expect("couldn't write to file");
        if pb.is_some() {
            pb.as_mut().unwrap().update(1).unwrap();
        }
    }
}
```

Reply:
I think we should refactor everything, because I designed interfaces without thinking about py-bindings. I suggest to have a one function `generate_csv(output_name: String, n: u64, k: u64, nas: u8, seed: u64, ds_type: Enum[groupby, join_big, join_big_na, join_medium, join_small])`

Inside this function we can initalize an instance of `RowGenerator` depends of the value of `ds_type` parameter

Reply:
Are we able to use the same function for both py-bindings and `main.rs`?

Reply:
I'll test that, I think the function might have to be in lib.rs and get called in main.rs though